diff --git a/scriptmodules/emulators/reicast.sh b/scriptmodules/emulators/reicast.sh
index a4135fe8..a02c2611 100644
--- a/scriptmodules/emulators/reicast.sh
+++ b/scriptmodules/emulators/reicast.sh
@@ -28,8 +28,15 @@ function sources_reicast() {
         gitPullOrClone "$md_build" https://github.com/reicast/reicast-emulator.git
     elif isPlatform "vero4k"; then
         gitPullOrClone "$md_build" https://github.com/reicast/reicast-emulator.git
+    elif isPlatform "tinker"; then
+        gitPullOrClone "$md_build" https://github.com/reicast/reicast-emulator.git
+        pushd "$md_build"
+            git checkout "TINKER_RETRO_PIE_OPTIONAL_NEW_REICAST_BRANCH"
+            git apply "TINKER_RETRO_PIE_OPTIONAL_NEW_REICAST/Makefile.patch"
+        popd
+        return
     else
-        gitPullOrClone "$md_build" https://github.com/RetroPie/reicast-emulator.git retropie
+        gitPullOrClone "$md_build" https://github.com/retropie/reicast-emulator.git retropie
     fi
     sed -i "s/CXXFLAGS += -fno-rtti -fpermissive -fno-operator-names/CXXFLAGS += -fno-rtti -fpermissive -fno-operator-names -D_GLIBCXX_USE_CXX11_ABI=0/g" shell/linux/Makefile
 }
@@ -40,8 +47,8 @@ function build_reicast() {
         make platform=rpi2 clean
         make platform=rpi2
     elif isPlatform "tinker"; then
-        make USE_GLES=1 USE_SDL=1 clean
-        make USE_GLES=1 USE_SDL=1
+        make platform=tinker-kms clean
+        make platform=tinker-kms
     else
         make clean
         make
@@ -54,7 +61,7 @@ function install_reicast() {
     if isPlatform "rpi"; then
         make platform=rpi2 PREFIX="$md_inst" install
     elif isPlatform "tinker"; then
-        make USE_GLES=1 USE_SDL=1 PREFIX="$md_inst" install
+        make platform=tinker-kms PREFIX="$md_inst" install
     else
         make PREFIX="$md_inst" install
     fi
